sdk/core/azure_core/examples/tables_cloud_configuration.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! Example showing how a Tables service client would use cloud configuration.
+//! This demonstrates the pattern that Tables and other services would follow.
+
+use azure_core::cloud::configurations;
+use azure_core::credentials::Secret;
+use azure_core::http::{ClientOptions, policies::BearerTokenCredentialPolicy};
+use azure_identity::ClientSecretCredential;
+use std::env;
+use std::sync::Arc;
+
+/// Example Tables client that demonstrates cloud configuration usage.
+#[derive(Debug)]
+pub struct ExampleTablesClient {
+    #[allow(dead_code)]
+    endpoint: String,
+    #[allow(dead_code)]
+    pipeline: Vec<Arc<dyn azure_core::http::policies::Policy>>,
+}
+
+impl ExampleTablesClient {
+    /// Create a new Tables client for Azure Public Cloud.
+    pub fn new_for_public_cloud(
+        account_name: &str,
+        credential: Arc<dyn azure_core::credentials::TokenCredential>,
+    ) -> Self {
+        let options = ClientOptions::default()
+            .with_cloud_config(configurations::azure_public_cloud())
+            .with_audience("https://storage.azure.com");
+        
+        Self::new_with_options(account_name, credential, options)
+    }
+
+    /// Create a new Tables client for Azure China Cloud.
+    pub fn new_for_china_cloud(
+        account_name: &str,
+        credential: Arc<dyn azure_core::credentials::TokenCredential>,
+    ) -> Self {
+        let options = ClientOptions::default()
+            .with_cloud_config(configurations::azure_china_cloud())
+            .with_audience("https://storage.azure.com");
+        
+        Self::new_with_options(account_name, credential, options)
+    }
+
+    /// Create a new Tables client for Azure US Government Cloud.
+    pub fn new_for_us_government_cloud(
+        account_name: &str,
+        credential: Arc<dyn azure_core::credentials::TokenCredential>,
+    ) -> Self {
+        let options = ClientOptions::default()
+            .with_cloud_config(configurations::azure_us_government_cloud())
+            .with_audience("https://storage.azure.com");
+        
+        Self::new_with_options(account_name, credential, options)
+    }
```

Comment:
Do NOT add methods for each cloud. The cloud configuration is only used if passed in via the options bag